### ISSUE TITLE
A translation error

### ISSUE THREAD

**Issue by ChicagoDuan** [ibm-openbmc/openbmc#304]
When the password complexity requirements are not met, the error message displays 'Error updating user 'admin' because the password was not accepted.' In this case, the prompt should indicate that the password does not meet the complexity requirements or specify which complexity requirement is not satisfied. The suggested English translation is: 'Error updating user 'admin' because the password complexity requirements are not met.

![image](https://github.com/ibm-openbmc/openbmc/assets/26787008/72ec19f7-4c64-4c5d-84ef-5e0c4139cef0)

**Comment by joseph-reynolds** [comment]
I agree.  I tried to fix the server side of this problem a few years ago via https://gerrit.openbmc.org/c/openbmc/bmcweb/+/27503, but the patch is abandoned because I ran out of time to work on it.

When I look at this patch, the remaining problems are:
1. The messages from Linux-PAM are hard-coded English and there is no translation support.  I didn't have a plan for what the WebUI is supposed to do with these messages (other then show them as toast messages).
2. It is difficult even for me to learn the exact set of messages.  Messages can come from any module named in the Linux-PAM /etc/pam.d "password" configuration (https://github.com/openbmc/openbmc/blob/master/meta-phosphor/recipes-extended/pam/libpam/pam.d/common-password).  The configuration stack changes over time, and different vendors may add modules, or enable pam_pwhistory (currently present but effectively disabled).  --> As it is now, Messages are typically from [pam_pwquality](https://github.com/libpwquality/libpwquality) but can also come from [pam_ipmicheck](https://github.com/openbmc/pam-ipmi/tree/master/src/pam_ipmicheck/pam_ipmicheck.c) such as  "Username %zu / Password %zu exceeds IPMI 16/20 limit". 
3. An effective way to solve this problem might be to first modify BMCWeb so it can reliably send the message `Error updating user 'admin' because the password was not accepted.`  The patch linked above is very close to doing this (providing properties so the WebUI can do this).  Maybe it can send the specific reasons (currently available as English-only strings) as additional data.
4. I'm not sure exactly what Redfish property should be used (per discussion in https://redfishforum.com/thread/246/message-send-patch-password-failure), or if the Password@Message.ExtendedInfo.Oem.OpenBMC.diagnosticMessages property (as-coded in the gerrit review) is acceptable.
5. The WebUI front-end (HTTP/Javascript code) would need to be enhanced to respond to this new Property from BMCWeb.

You are welcome to pick up this patch and work on it.

**Comment by mzipse** [comment]
@ChicagoDuan , would you be willing to pick this patch up, modify appropriately to address some of Joseph's concerns and then push it upstream?  Once the backend work is upstreamed, I can get the BMC GUI team to make a corresponding change.

**Comment by ChicagoDuan** [comment]
> @ChicagoDuan , would you be willing to pick this patch up, modify appropriately to address some of Joseph's concerns and then push it upstream? Once the backend work is upstreamed, I can get the BMC GUI team to make a corresponding change.

Sure！@lxwinspur George will pick this patch up.

**Comment by ChicagoDuan** [comment]
Hi @mzipse ,
We found that this issue is not related to the patch submitted by @joseph-reynolds , but is caused by a string matching error in the webui. The following commit has fixed this issue:
https://github.com/ibm-openbmc/webui-vue/commit/9f3ae79bd59d0a4c5867ceef744d171edefd352e

**Comment by mzipse** [comment]
This fix is merged into the next 1020, 1030, 1050 service packs, and 1060.

**Comment by mzipse** [comment]
Closing this issue now.
